src/spectral.jl

```diff
@@ -264,3 +264,25 @@ function contract(nbt::Nonbacktracking, edgespace::Vector)
     contract!(y,nbt,edgespace)
     return y
 end
+
+"""spectral_distance(G₁, G₂ [, k])
+Compute the spectral distance between undirected graphs
+G₁ and G₂ using the top k greatest eigenvalues. If k is
+ommitted, uses k = min(|G₁|,|G₂|).
+
+For further details, please refer to:
+
+JOVANOVIC, I.; STANIC, Z., 2014. Spectral Distances of
+Graphs Based on their Different Matrix Representations
+"""
+function spectral_distance(G₁::Graph, G₂::Graph, k::Integer)
+  A₁ = adjacency_matrix(G₁)
+  A₂ = adjacency_matrix(G₂)
+
+  λ₁ = k < nv(G₁)-1 ? eigs(A₁, nev=k, which=:LR)[1] : eigvals(full(A₁))[end:-1:end-(k-1)]
+  λ₂ = k < nv(G₂)-1 ? eigs(A₂, nev=k, which=:LR)[1] : eigvals(full(A₂))[end:-1:end-(k-1)]
+
+  sumabs(λ₁ - λ₂)
+end
+
+spectral_distance(G₁::Graph, G₂::Graph) = spectral_distance(G₁, G₂, min(nv(G₁), nv(G₂)))
```

Comment:
Hi @CarloLucibello, what is the construct that you have in mind? All these distance definitions are new to me, I will find time to read the paper again carefully and see if there is indeed a hidden `n1 == n2` statement somewhere.